src/js/controller/next-item-promise.js

```diff
@@ -0,0 +1,60 @@
+
+export class ItemPromise {
+
+    constructor (index, model, api) {
+        this.index = index;
+        this.model = model;
+        this.api = api;
+        this.promise = new Promise((resolve, reject) => {
+            this.resolve = resolve;
```

Comment:
i _love_ this approach to encapsulating the promise without a single overcomplex function

Reply:
hoisting those is not great, but one promise per class instance keeps it under control...